### PR TITLE
Use IP addresses for OpenStack nodes

### PR DESCRIPTION
**Pull request by tomassedovic**
Unlike other cloud providers, OpenStack VMs are not able to resolve each
other by their names. If you try to run the playbooks against nodes
without a pre-created /etc/hosts or a DNS that provides the hostname/ip
resolution, it will fail.

By setting the `openshift_hostname` variable to each node's IP address,
we're able to deploy a functional cluster without running a custom DNS.

It is still possible to provide an external server with nsupdate keys
and have it be populated, but that is no longer a hard requirement.

### CODE REVIEW

File: playbooks/openstack/advanced-configuration.md [hunk 1]
```diff
@@ -47,44 +47,42 @@ Otherwise, even if there are differences between the two versions, installation
 
 ## Accessing the OpenShift Cluster
 
-### Use the Cluster DNS
+### Configure DNS
 
-In addition to the OpenShift nodes, we created a DNS server with all
-the necessary entries. We will configure your *Ansible host* to use
-this new DNS and talk to the deployed OpenShift.
+OpenShift requires two DNS records to function fully. The first one poinst to
```
Comment (bogdando) [inline playbooks/openstack/advanced-configuration.md]: a typo: points

File: playbooks/openstack/advanced-configuration.md [hunk 2]
```diff
@@ -47,44 +47,42 @@ Otherwise, even if there are differences between the two versions, installation
...
+the master/load balancer and provides the UI/API access. The other one is a
+wildcard domain that resolves app route requests to the infra node.
 
-First, get the DNS IP address:
+If you followed the default installation from the README section, there is no
+DNS configured.  We can add two entries to your `/etc/hosts` to do a quick
```
Comment (bogdando) [inline playbooks/openstack/advanced-configuration.md]: 'You should add'

File: playbooks/openstack/advanced-configuration.md [hunk 3]
```diff
@@ -47,44 +47,42 @@ Otherwise, even if there are differences between the two versions, installation
...
-to pings.
+First, run the `openstack server list` command and note the floating IP
+addresses of the *master* and *infra* nodes (we will use `10.40.128.130` for
+master and `10.40.128.134` for infra here).
 
-Next, edit your `/etc/resolv.conf` as root and put `nameserver DNS_IP` as your
-**first entry**.
-
-If your `/etc/resolv.conf` currently looks like this:
+Then add the following entries to your `/etc/hosts`:
```
Comment (bogdando) [inline playbooks/openstack/advanced-configuration.md]: let's please clarify 'your' as 'ansible admin/control node'

File: playbooks/openstack/advanced-configuration.md [hunk 4]
```diff
@@ -47,44 +47,42 @@ Otherwise, even if there are differences between the two versions, installation
...
-; generated by /usr/sbin/dhclient-script
-search openstacklocal
-nameserver 192.168.0.3
-nameserver 192.168.0.2
+10.40.128.130 console.openshift.example.com
+10.40.128.134 cakephp-mysql-example-test.apps.openshift.example.com
 ```
 
-Change it to this:
+I.e. point the cluster domain (as defined in the
```
Comment (bogdando) [inline playbooks/openstack/advanced-configuration.md]: please use "This points the cluster"

File: playbooks/openstack/sample-inventory/inventory.py [hunk 5]
```diff
@@ -79,10 +79,18 @@ def build_inventory():
...
         # TODO(shadower): what about multiple networks?
         if server.private_v4:
             hostvars['private_v4'] = server.private_v4
+            # NOTE(shadower): Yes, we set both hostname and IP to the private
+            # IP address for each node. OpenStack doesn't resolve nodes by
+            # name at all, so using a hostname here would require an internal
+            # DNS which would complicate the setup and potentially introduce
+            # performance issues.
+            hostvars['openshift_ip'] = server.private_v4
+            hostvars['openshift_hostname'] = server.private_v4
```
Comment (bogdando) [inline playbooks/openstack/sample-inventory/inventory.py]: could we please as well explicitly define `vars['openshift_public_hostname']`? Just following the 'explicit is better than explicit' rule.

**Comment by bogdando** [comment]
And 'Prevent cloud-init updates of hostname/fqdn' is not needed as well, I suppose

**Comment by tomassedovic** [comment]
@bogdando thanks for suggestions! I've done the text and inventory fixes, but I'd like to remove the cloud-init etc. code in another pull request.

This one just updates the public interfaces (inventory and docs), but there's a ton of stuff we'll want to drop (dns in the heat templates, role defaults, etc.) so I'd prefer to do that separately.

**Comment by sdodson** [comment]
flakes

**Pull request merged**